### ISSUE TITLE
Inspection of openrefine objects in preview

### ISSUE THREAD

**Issue by msaby** [OpenRefine/OpenRefine#2120]
**Is your feature request related to a problem or area of OpenRefine? Please describe.**

In preview mode, when the result of an expression is an openrefine object (row, record...), it could be useful to give the user some hints about the properties of this object.

For example if the result is "cell", the user should see that "cell.recon" is a valid expression, then he could fix his expression by entering "cell.recon", and the system would inform him that "cell.recon.match" is valid, and at the end he could type a complex expression like "cell.recon.match.id"

**Describe the solution you'd like**

in preview mode : a little icon just before any Openrefine object (like cell, row, row.cell...), or a highlight + a tooltip showing the properties of the object when the mouse pass over the object.

**Describe alternatives you've considered**

**Additional context**

**Comment by msaby** [comment]
It could be handled both in expression editor and in the preview behind.

**Comment by thadguidry** [comment]
**DESIGN NOTE:**  This issue has a general theme that is similar to #443 and discussions and other issues mentioned in #1088

**Comment by wetneb** [comment]
What is the intended meaning of the `design_notes` tag by the may?

**Comment by thadguidry** [comment]
@wetneb I've been using that to help narrow discussion areas with any future design changes in OR.  Design being a huge elephant effort potentially.  You can think of it as notes against the BEAUTY QUEEN effort that might/should happen in the future.

Its more than an enhancement label... its "This is an enhancement, that needs design work, possibly lengthy discussions about the overall design of how OR works in areas or how it could work differently.  It is not only about UI/UX however as we know that sometimes backend changes will be needed to support the design that is proposed to being reshaped."

For example, if you look at David's reply in my #443 he mentions a few things to think about from an overall design.

An alternative to this generic label, might be a Projecst link (not Milestone link) to the [UI Improvements 2019](https://github.com/OpenRefine/OpenRefine/projects/2) and instead rename that to something called "BEAUTY QUEEN"... anyways its not always about beauty or the UI/UX but that of potentially changing the design of how OR works in areas.  Make sense ?  We need to use Projects more and probably begin linking issues them.

I generally want a way to get quick perspective listing on the future of OR's design.  And right now they ideas are sometimes deep inside an issue itself and when I spot that...I label it with "design note".  Maybe it should be a Project link to "BEAUTY QUEEN" or something meaningful like "DESIGN REFACTOR"... thoughts?

**Comment by wetneb** [comment]
In my opinion most issues need a design effort to be addressed, so this tag could be applied to most enhancement issues…

I am not against using GitHub projects more. For me a project should correspond to a particular funded effort (showing the tasks within the scope of that particular effort), or other initiatives focused on a specific topic. I do not think it is useful to gather all UI improvement issues in a given project: tags can already do that.

Also, again I am really not in favour of using this "Beauty Queen" phrase, I think it is really not professional. Can we please stop using it?

**Comment by thadguidry** [comment]
Agreed on the UI "Beauty Queen" phrase. Dropping it, thanks for pointing out.

A project doesn't have to correspond to a funded effort I would say.  They can be unfunded and someone volunteers to drive the effort.  But agreed Projects should correspond to a scope of effort or initiative.

Part of the rationale I had was also to say to a UI/UX designer if ever brought in... "here's a list of design considerations that we have collected".  Just "enhancement" tags doesn't help him/her since the issue might be only GREL syntax for instance on many of those issues and nothing to do with UI/UX.

I generally feel we are doing a poor job of planning for the future of OpenRefine.  Don't you feel that also? I don't think we surface a good picture to present to a potential UI/UX designer.  This was something we did similarly when James Home and other Google designers came in to help us refactor the design between the evolution of Gridworks -> Google Refine.  I feel we are missing that picture, which is fragmented into a thousands pieces across (your papers, my notes, our Phase 1/2 Word docs, our Wiki, elsewhere)

**Comment by wetneb** [comment]
Yes issues are too fragmented, we would need more structure on top of it to see where we are heading. The problem is that by default the project is kind of running in "keeping-the-lights-on" mode: responding to bug reports, reviewing pull requests, with no bandwidth to plan big changes.

For that mode of operation there is no big need for a lot more structure: we just fix small things on the side.

To plan for the future and propose big changes, perhaps the dev mailing list is better suited? I don't know. I am trying to get more familiar with WMDE's way of working, where they have various sorts of tickets (big overarching ones describing the target user stories that they are aiming for, and then more granular ones to detail the changes required).

**Comment by thadguidry** [comment]
Yes, the dev mailing list for announcements and general tracking.  Let me compose a continuation there...
